modules/biobank/php/biobank.class.inc

```diff
@@ -0,0 +1,117 @@
+<?php declare(strict_types=1);
+/**
+ * This class features the code that enables access to the LORIS Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Page
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+namespace LORIS\biobank;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * This class features the code that enables access to the LORIS Biobank Module.
+ *
+ * PHP Version 7.2
+ *
+ * @category   Page
+ * @package    Loris
+ * @subpackage Biobank
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.text GPLv3
+ * @link       http://www.github.com/aces/Loris/
+ */
+class Biobank extends \NDB_Menu_Filter
+{
+    /**
+     * Determine whether the user has permission to view this page.
+     *
+     * @param User $user to be checked.
+     *
+     * @return bool whether the user has access.
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return $user->hasPermission('biobank_access');
+    }
+
+    /**
+     * Set up the variables required by for construction of page
+     *
+     * @return void
+     */
+    function _setupVariables()
+    {
+        $this->skipTemplate = true;
+        $this->AjaxModule   = true;
+    }
+
+    // TODO: Move this to another class.
+    /**
+     * Allows files to be downloaded
+     *
+     * @param \SplFileObject $file File to be downloaded
+     *
+     * @return ResponseInterface the outgoing PSR7 response.
+     */
+    function downloadFile(\SplFileObject $file) : ResponseInterface
+    {
+        if (!$user->hasPermission('biobank_specimen_view')) {
+            throw new \Forbidden('Permission to download file is denied');
+        }
+
+        $fileName = basename($file);
+        $config   = \NDB_Config::singleton();
+        $path     = $config->getSetting('mediaPath');
+        $filePath = $path . $fileName;
```

Comment:
We don't do it most places, but it would still be a good idea to throw an exception if `$path` doesn't end with a slash here. (I'm assuming `$fileName` doesn't start with one..)

Reply:
@driusan Maybe it would be better to that upstream in my PR? https://github.com/aces/Loris/pull/4023 Otherwise we'll have to do it all over

Reply:
we should do it wherever we can most consistently, I just noticed it here because it's new code.. but I don't see how this instance can be solved elsewhere in the code. It's directly using string concatenation with a getSetting value, and it can't be done in getSetting because it's too generic.

Reply:
Resolving while John develops the new file downloader.